homeassistant/components/yamaha/media_player.py

```diff
@@ -143,46 +100,30 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Yamaha platform."""
-    # Keep track of configured receivers so that we don't end up
-    # discovering a receiver dynamically that we have static config
-    # for. Map each device from its zone_id .
-    known_zones = hass.data.setdefault(DOMAIN, {KNOWN_ZONES: set()})[KNOWN_ZONES]
-    _LOGGER.debug("Known receiver zones: %s", known_zones)
-
-    # Get the Infos for configuration from config (YAML) or Discovery
-    config_info = YamahaConfigInfo(config=config, discovery_info=discovery_info)
-    # Async check if the Receivers are there in the network
-    try:
-        zone_ctrls = await hass.async_add_executor_job(_discovery, config_info)
-    except requests.exceptions.ConnectionError as ex:
-        raise PlatformNotReady(f"Issue while connecting to {config_info.name}") from ex
-
-    entities = []
-    for zctrl in zone_ctrls:
-        _LOGGER.debug("Receiver zone: %s serial %s", zctrl.zone, zctrl.serial_number)
-        if config_info.zone_ignore and zctrl.zone in config_info.zone_ignore:
-            _LOGGER.debug("Ignore receiver zone: %s %s", config_info.name, zctrl.zone)
-            continue
-
-        assert config_info.name
-        entity = YamahaDeviceZone(
-            config_info.name,
-            zctrl,
-            config_info.source_ignore,
-            config_info.source_names,
-            config_info.zone_names,
-        )
-
-        # Only add device if it's not already added
-        if entity.zone_id not in known_zones:
-            known_zones.add(entity.zone_id)
-            entities.append(entity)
+    if config.get(CONF_HOST):
+        if hass.config_entries.async_entries(DOMAIN) and config[CONF_HOST] not in [
+            entry.data[CONF_HOST] for entry in hass.config_entries.async_entries(DOMAIN)
+        ]:
+            _LOGGER.error(
+                "Configuration in configuration.yaml is not supported anymore. "
+                "Please add this device using the config flow: %s",
+                config[CONF_HOST],
+            )
         else:
-            _LOGGER.debug(
-                "Ignoring duplicate zone: %s %s", config_info.name, zctrl.zone
+            _LOGGER.warning(
+                "Configuration in configuration.yaml is deprecated. Use the config flow instead"
             )
 
-    async_add_entities(entities)
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+                )
+            )
+    else:
+        _LOGGER.error(
```

Comment:
Start a discovery instead and for each discovered receiver launch an import flow.